gemd/entity/bounds/categorical_bounds.py

```diff
@@ -70,6 +70,53 @@ def contains(self, bounds: Union[BaseBounds, "BaseValue"]) -> bool:
 
         return bounds.categories.issubset(self.categories)
 
+    def union(self,
+              *others: Union["CategoricalBounds", "CategoricalValue"]
+              ) -> "CategoricalBounds":
+        """
+        Return the union of this bounds and other bounds.
+
+        The others list must also be Categorical Bounds or Values.
+
+        Parameters
+        ----------
+        others: Union[CategoricalBounds, CategoricalValue]
+            Other bounds or value objects to include.
+
+        Returns
+        -------
+        CategoricalBounds
+            The union of this bounds and the passed bounds
+
+        """
+        from gemd.entity.value.categorical_value import CategoricalValue
+
+        if any(not isinstance(x, (CategoricalBounds, CategoricalValue)) for x in others):
+            misses = {type(x).__name__
+                      for x in others
+                      if not isinstance(x, (CategoricalBounds, CategoricalValue))}
+            raise TypeError(f"union requires consistent typing; found {misses}")
```

Comment:
The error for this (and the other implementations) should say which type was expected. So here for example, "union requires consistent typing; expected categorical, found {misses}"